tools/nl/embeddings/build_embeddings.py

```diff
@@ -31,19 +36,44 @@
 flags.DEFINE_string('catalog', '',
                     'A dict of user provided embeddings/model catalog')
 
+flags.DEFINE_string(
+    'additional_catalog_path', '',
+    'Path to an additional catalog yaml file. Can be a local or a GCS path')
+
+
+def _init_logger():
+  # Log to stdout for easy redirect of the output text.
+  # This enables the logs to be captured by the admin tool.
+  logger = logging.getLogger()
+  logger.setLevel(logging.INFO)
+  handler = logging.StreamHandler(sys.stdout)
+  handler.setLevel(logging.INFO)
+  logger.addHandler(handler)
+
 
 def main(_):
+  _init_logger()
+
   # Get embeddings_name
   embeddings_name = FLAGS.embeddings_name
   output_dir = FLAGS.output_dir
+
+  # If additional catalog path is provided and embeddings name is not specified, extract the embeddings name from the catalog.
```

Comment:
Can this too happen upstream in "admin.py".  So the interface of this tool is:  catalog_path and embeddings_name.

Specifically, the decision to pick the first index-name in the additional-catalog-file is a very custom-dc specific decision that's not generally applicable?

Reply:
The separation would be nice. However, once we have data docker, admin/html.py will be deprecated and the tool will be called by a shell script in docker.

Reply:
Hmm, if it will have this fixed name - https://github.com/datacommonsorg/import/pull/332/files#diff-5ea0ff46203687a6da5298b11caf748eeb36522e1ee79ec018d18bdee6693369R25 - wonder if the caller (html.py or that script) should pass that in?  Anyway the file there is constants shared with this repo.

Reply:
That's a good point. Updated code to pass embeddings name from admin (and in the future from docker). Also removed code that extracted the first name from the catalog.

Reply:
Much better, thanks!